src/net/server/channel/Channel.java

```diff
@@ -485,4 +477,8 @@ public boolean setDojoProgress(int dojoMapId) {
     public long getDojoFinishTime(int dojoMapId) {
         return dojoFinishTime[getDojoSlot(dojoMapId)];
     }
+
+    public MapleEvent getEvent() {
+        return event;
+    }
```

Comment:
I actually want to remove this - the legacy GM events still use this function but I plan on updating the other events to follow the new core structure.